index.html

```diff
@@ -302,18 +304,27 @@ <h3>
           {{DOMString}} "<code>navigation</code>".
           </li>
           <li>The <dfn id='dom-PerformanceNavigationTiming-workerStart'><code>
-            workerStart</code></dfn> attribute MUST return the time immediately
-            before the user agent <a data-cite=
-            "service-workers#run-service-worker">ran the worker</a> (if the
-            <a>current document</a> has an <a data-cite=
-            "service-workers#dfn-containing-service-worker-registration">active
-            service worker registration</a> [[SERVICE-WORKERS]]) required to
-            service the request, or if the worker was already available, the
-            time immediately before the user agent <a data-cite=
-            "service-workers#fetchevent">fired an event named `fetch`</a> at
-            the <a data-cite="service-workers#dfn-active-worker">active
-            worker</a>. Otherwise, if there is no active worker this attribute
-            MUST return zero.
+            workerStart</code></dfn> attribute getter does the following:
+            <ul>
+              <li>If the <a>current document</a> has no <a data-cite=
+                "service-workers#dfn-containing-service-worker-registration">active
+                service worker registration</a> [[SERVICE-WORKERS]], this attribute MUST
+                return zero.</li>
+              <li>If there were redirects, this attribute MUST return a {{DOMHighResTimeStamp}}
```

Comment:
I still think these if statements are a bit hard to follow. In this if statement, we only see `If there were redirect` but the condition is more than that: there were redirects and we had to run a worker? Or am I missing something?

Reply:
In other words it seems this should be split into two, similar to the next two ifs.

Reply:
Hm, to simplify it maybe something like:

```
If there were redirects, this attribute MUST return a {{DOMHighResTimeStamp}}
with the `workerStart` time recorded by the first request in final same-origin redirect chain.
````

Or is that too much of a circular dependency?